blivet/devices/device.py

```
@@ -71,6 +71,13 @@ class Device(util.ObjectID, metaclass=SynchronizedMeta):
     _packages = []
     _external_dependencies = []
 
+    # Read-only methods/properties exempted from blivet_lock to prevent
+    # deadlock when __str__ is called during logging format (which holds
+    # the logging handler lock). These are safe to call without the lock
+    # since they only read state and do not modify it.
+    _unsynchronized_methods = ['__str__', '_to_string', '_get_name',
+                               'type', 'children']
```

Ruff (0.15.21)

[warning] Mutable default value for class attribute

(RUF012)